### ISSUE TITLE
Add Extended Transfer Tool Options

### ISSUE THREAD

**Issue by NickolausDS** [globus-gladier/gladier-tools#59]
The Transfer Tool is simplistic in nature so it's easy to use. But it's missing a lot of options that are typically very useful, such as `skip_source_errors` and `notify_on_succeeded`. Adding more inputs would fix this, but would also bloat the main `$.input` for all transfers. There are a few routes we could take to address this: 

* Add only select options we deem the most useful
* Add all options as default flow options in the definition via expressions `("skip_source_errors.=": "getattr(input.skip_source_errors, False"`
    * Potentially mis-leading, for example if `$.input.skip_source_errors` wasn't intended as input for this transfer tool. 
* Add a second extended transfer tool which contains all options
* Do nothing, and let people add their own tools if the simple one doesn't fit.

**Comment by ravescovi** [comment]
I like option2. Expanding the current tool with more defaults. That would
be back-compatible and still open the new flags.

**Comment by NickolausDS** [comment]
If we used option 2 and additionally changed the input location to something like `$.input.transfer`, it would avoid the drawback. Not backwards compatible, but probably a better long-term solution.

**Comment by ravescovi** [comment]
are we suggesting we leave on the top level what is already there and move to `.transfer` the new inputs?
Wouldn't that be outdated when we move all tools into their own `input.toolX`?

**Comment by NickolausDS** [comment]
No, I was suggesting moving everything. So all inputs for the transfer tool would be expected under `$.input.transfer`. I'm not certain how that would work for aliasing, I would need to take another look. But I'd expect each to be a nested block for each aliased tool where each block contained only transfer items. So input wouldn't be mixed between the Transfer tools on the top level `$.input`
